pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go

```diff
@@ -388,6 +388,8 @@ func (t *ResolvedPipelineTask) skip(facts *PipelineRunFacts) TaskSkipStatus {
 		skippingReason = v1beta1.PipelineTimedOutSkip
 	case t.skipBecausePipelineRunTasksTimeoutReached(facts):
 		skippingReason = v1beta1.TasksTimedOutSkip
+	case t.skipBecauseEmptyArrayInMatrixParams(facts):
+		skippingReason = v1beta1.EmptyArrayInMatrixParams
```

Comment:
a similar check needs to be added to [IsFinallySkipped](https://github.com/tektoncd/pipeline/blob/57577006b054c3bc45f3f668dfba890533322100/pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go#L513)

Reply:
Added to [IsFinallySkipped](https://github.com/tektoncd/pipeline/blob/57577006b054c3bc45f3f668dfba890533322100/pkg/reconciler/pipelinerun/resources/pipelinerunresolution.go#L513) as well